interfaces/cython/cantera/yaml2ck.py

```diff
@@ -0,0 +1,1047 @@
+    """
+    Creates the header text for all output files
+
+    :param solution:
+        The `solution` object being converted
+    :param max_width:
+        The maximum width of lines before they start to wrap
+    """
+    text = []
+
+    if "description" in solution.input_header:
+        text_wrap = lambda t: fill(
+            t,
+            width=max_width,
+            break_long_words=False,
+            break_on_hyphens=False,
+            initial_indent="! ",
+            subsequent_indent="!   ",
+        )
+
+        text = solution.input_header["description"].splitlines()
+        text = [text_wrap(line) for line in text]
+    else:
+        note = ""
+
+    ct_note = (
+        f"! Chemkin file was converted from a Cantera {ct.__version__} "\
+        "`solution` object"
+    )
+
+    has_ct_note = False
+    for n, line in enumerate(text):
+        if "! Chemkin file was converted from a Cantera" in line:
+            text[n] = ct_note
+            has_ct_note = True
+            break
+
+    if not has_ct_note:
+        text.append(f"!\n{ct_note}\n! \n\n")
+
+    return "\n".join(text)
+
+
+def build_species_text(solution, max_width=80, sort_elements=True, 
+                       sort_species=True):
+    """
+    Creates element and species declarations
+
+    :param solution:
+        The `solution` object being converted
+    :param max_width:
+        The maximum width of lines before they start to wrap
+    :param sort_elements:
+        A boolean flag to indicate if elements should be sorted
+    :param sort_species:
+        A boolean flag to indicate if species should be sorted
+    """
+
+    def build_elements(elements, atomic_weights):
+        if sort_elements:  # Sort elements by atomic weight
+            elements = [c for _, c in sorted(zip(atomic_weights, elements))]
+
+            if sort_elements:
+                elements = elements[::-1]
+
+        elements_text = fill(
+            "  ".join(elements),
+            width=max_width,
+            break_long_words=False,
+            break_on_hyphens=False,
+        )
+
+        text = ["ELEMENTS"]
+        text.append(f"{elements_text}")
+        text.append("END\n\n")
+
+        return "\n".join(text)
+
+    def build_species(solution_species):
+        # get notes from all species
+        note = {}
+        for species in solution_species:
+            species_name = species.input_data["name"]
+            if "note" in species.input_data:
+                note[species_name] = species.input_data["note"]
+            else:
+                note[species_name] = ""
+
+        # sort names
+        species_names = [species.name for species in solution_species][::-1]
+        if sort_species:  # sort solution_species by molecular weight
+            MW = solution.molecular_weights
+            species_names = [c for _, c in sorted(zip(MW, species_names))]
+
+            if sort_species:
+                species_names = species_names[::-1]
```

Comment:
Same comment here as the elements about reversing the list and the conditional.

Reply:
I prefer my mechanisms to go from heaviest to lightest when reading species, so that's what I put in, or at least that's what it used to be before I made changes.

Reply:
The mistake was not also having this in my thermodynamics section, which I've fixed.